### DISCUSSION TITLE
how to upgrade JNI version?

### DISCUSSION THREAD

**Discussion by anlexN** [android/ndk#2009] [General]
I can not find imformation in the google, github discussions and ndk docs.

**Comment by enh-google** [comment]
the return value from JNI_OnLoad() is how you tell the system what JNI version you understand. see https://developer.android.com/training/articles/perf-jni#native-libraries for an example.

you can use the GetVersion() JNI function to query what the system actually supports.

**Reply by anlexN** [reply]
I have used `GetVersion()`, it say `0x00010006`.
jni 1.6 is very old. how to upgrade?

**Reply by enh-google** [reply]
"upgrade" doesn't make any sense --- you get what you get.

**Reply by anlexN** [reply]
I need JNI latest syntax and performance. I can not upgrade? android ndk lock in JNI 1.6?

**Reply by DanAlbert** [reply]
The NDK's got nothing to do with this. The device has JNI 1.6.

**Reply by DanAlbert** [reply]
Are you confusing JNI version and _Java_ version? They're not related. You can use modern Java syntax regardless of what version of JNI the device supports.
